crates/ty_python_semantic/src/types.rs

```diff
@@ -1528,6 +1568,11 @@ impl<'db> Type<'db> {
             }
             (Type::Tuple(left), Type::Tuple(right)) => left.is_equivalent_to(db, right),
             (Type::Callable(left), Type::Callable(right)) => left.is_equivalent_to(db, right),
+            // A fully static typevar is always equivalent to itself, and not any other typevar,
+            // since there is no guarantee that they will be specialized to the same type. (This is
+            // true even if both typevars are bounded by the same final class, since you can
+            // specialize the typevars to `Never` in addition to that final class.)
+            (Type::TypeVar(first), Type::TypeVar(second)) => first == second,
```

Comment:
I didn't do this in my PR, but maybe we should add a test for this? Apparently we don't have one.